File: baekjoon/10866/Lactea98/10866.cpp
#include <iostream>
#include <deque>
#include <cstring>

using namespace std;

int main(){
    int num, tmp;
    char command[11];
    deque<int> d;

    scanf("%d", &num);
    while(num--){
        scanf("%s", command);

        if(!strcmp(command, "push_front")){
            scanf("%d", &tmp);
            d.push_front(tmp);
        }
        else if(!strcmp(command, "push_back")){
            scanf("%d", &tmp);
            d.push_back(tmp);
        }
        else if(!strcmp(command, "pop_front")){
            if(d.size() == 0) printf("-1\n");
            else {
                printf("%d\n", d[0]);
                d.pop_front();
            }
        }
        else if(!strcmp(command, "pop_back")){
            if(d.size() == 0) printf("-1\n");
            else {
                printf("%d\n", d[d.size() - 1]);
                d.pop_back();
            }
        }
        else if(!strcmp(command, "size")) printf("%d\n", d.size());
        else if(!strcmp(command, "empty")) printf("%d\n", (d.empty() ? 1 : 0));
        else if(!strcmp(command, "front")){
            if(d.size() == 0) printf("-1\n");
            else printf("%d\n", d[0]);
        }
        else if(!strcmp(command, "back")){
            if(d.size() == 0) printf("-1\n");
            else printf("%d\n", d[d.size() - 1]);
        }
    }
}

File: baekjoon/data_structure/10866/10866.py
import sys
from collections import deque

sys.stdin = open("baekjoon\data_structure\\10866\input.txt")

d = deque()
n = int(sys.stdin.readline())

for _ in range(n):
    cmd = list(sys.stdin.readline().strip().split(" "))
    
    if cmd[0] == 'push_front':
        d.appendleft(cmd[1])
        
    elif cmd[0] == 'push_back':
        d.append(cmd[1])
        
    elif cmd[0] == 'pop_front':
        if d:
            tmp = d.popleft()
            print(tmp)
        else:
            print("-1")
            
    elif cmd[0] == 'pop_back':
        if d:
            tmp = d.pop()
            print(tmp)
        else:
            print("-1")
            
    elif cmd[0] == 'size':
        print(len(d))
        
    elif cmd[0] == 'empty':
        if d:
            print(0)
        else:
            print(1)
            
    elif cmd[0] == 'front':
        if d:
            print(d[0])
        else:
            print("-1")
        
    elif cmd[0] == 'back':
        if d:
            print(d[len(d)-1])
        else:
            print("-1")
    